build error with MPI code

## Expected Behavior

The MPI parcelport code used to compile just fine

## Actual Behavior

Parcelport code is now failing to build due to a type mismatch.

```bash
/home/users/khuck/src/phylanx/tools/buildbot/src/hpx/hpx/plugins/parcelport/mpi/sender.hpp: In constructor ‘hpx::parcelset::policies::mpi::sender::sender()’:
/home/users/khuck/src/phylanx/tools/buildbot/src/hpx/hpx/plugins/parcelport/mpi/sender.hpp:41:30: error: invalid conversion from ‘int’ to ‘MPI_Request {aka ompi_request_t*}’ [-fpermissive]
           , next_free_tag_(-1)
                              ^
make[2]: *** [src/CMakeFiles/hpx.dir/__/plugins/parcelport/mpi/parcelport_mpi.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/CMakeFiles/hpx.dir/all] Error 2
make: *** [all] Error 2
```

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1.  cmake -DCMAKE_CXX_COMPILER=/packages/gcc/7.3.0/bin/g++ -DCMAKE_C_COMPILER=/packages/gcc/7.3.0/bin/gcc -DCMAKE_BUILD_TYPE=Release -DBOOST_ROOT=/home/users/khuck/src/phylanx/tools/buildbot/build-cyclops-ppc64le-Linux-gcc/boost-1.65.1 -DHPX_WITH_MALLOC=system -DHWLOC_ROOT= -DCMAKE_INSTALL_PREFIX=. -DHPX_WITH_THREAD_IDLE_RATES=ON -DHPX_WITH_PARCELPORT_MPI=ON -DHPX_WITH_PARCEL_COALESCING=OFF -DHPX_WITH_TOOLS=OFF -DHPX_WITH_TESTS=ON -DHPX_WITH_EXAMPLES=OFF -DHPX_WITH_THREAD_MANAGER_IDLE_BACKOFF=OFF -DHPX_WITH_APEX=TRUE -DAPEX_WITH_ACTIVEHARMONY=TRUE -DACTIVEHARMONY_ROOT=/usr/local/packages/activeharmony/4.6.0-ppc64le -DAPEX_WITH_OTF2=TRUE -DOTF2_ROOT=/usr/local/packages/otf2/2.0-ppc64le -DAPEX_WITH_PAPI=TRUE -DPAPI_ROOT=/usr/local/packages/papi/5.5.1 -DHPX_WITH_APEX_NO_UPDATE=FALSE -DHPX_WITH_APEX_TAG=develop -DHWLOC_ROOT=/usr/local/packages/hwloc/1.8 -DHPX_WITH_RDTSCP=OFF -DHPX_WITH_DEPRECATION_WARNINGS=Off -DHPX_WITH_MAX_CPU_COUNT=160 /home/users/khuck/src/phylanx/tools/buildbot/src/hpx
  1.  make
  1.

## Specifications

... Please describe your environment

  - HPX Version:  current master
  - Platform (compiler, OS):  all platforms, gcc 7 with openmpi 4.0

--- a/hpx/plugins/parcelport/mpi/sender.hpp
+++ b/hpx/plugins/parcelport/mpi/sender.hpp
@@ -37,7 +37,7 @@ namespace hpx { namespace parcelset { namespace policies { namespace mpi
         using mutex_type = hpx::lcos::local::spinlock;
 
         sender()
-          : next_free_tag_request_(-1)
+          : next_free_tag_request_((MPI_Request)(-1))
           , next_free_tag_(-1)
         {
         }

